[8.16]  Updates CSPM guides to include agentless option (#5863)
elastic/security-docs: docs/cloud-native-security/cspm-get-started-aws.asciidoc

[[cspm-setup]]
== Set up CSPM for AWS

> You can set up CSPM for AWS either by enrolling a single cloud account, or by enrolling an organization containing multiple accounts. Either way, first you will add the CSPM integration, then enable cloud account access. Two deployment technologies are available: agentless, and agent-based. <<cspm-aws-agentless, Agentless deployment>> allows you to collect cloud posture data without having to manage the deployment of an agent in your cloud. <<cspm-aws-agent-based, Agent-based deployment>> requires you to deploy and manage an agent in the cloud account you want to monitor. 

Reviewer: Not sure if you're specifically referring to Elastic's Agent here or agents in a general sense. If it's the former, this change applies across all related pages. 

```suggestion
You can set up CSPM for AWS either by enrolling a single cloud account, or by enrolling an organization containing multiple accounts. Either way, first you will add the CSPM integration, then enable cloud account access. Two deployment technologies are available: agentless, and agent-based. <<cspm-aws-agentless, Agentless deployment>> allows you to collect cloud posture data without having to manage the deployment of {agent} in your cloud. <<cspm-aws-agent-based, Agent-based deployment>> requires you to deploy and manage {agent} in the cloud account you want to monitor. 
```

[discrete]
[[cspm-aws-agentless]]
== Agentless deployment 
beta::[]

. From the Elastic Security *Get started* page, click *Add integrations*.
. Search for `CSPM`, then click on the result.
. Click *Add Cloud Security Posture Management (CSPM)*.
. Select *AWS*, then either *AWS Organization* to onboard multiple accounts, or *Single Account* to onboard an individual account.
. Give your integration a name that matches the purpose or team of the AWS account/organization you want to monitor, for example, `dev-aws-account`.

> . Click **Advanced options** then select **Agentless (BETA)**.

Reviewer: ```suggestion
. Click **Advanced options**, then select **Agentless (BETA)**.
```

. Next, you'll need to authenticate to AWS. Two methods are available:

> .. Option 1: Direct access keys / CloudFormation (Recommended). Under **Preferred method** select **Direct access keys**. Expand the **Steps to Generate AWS Account Credentials** section, then follow the instructions that appear to automatically create the necessary credentials using CloudFormation.

Reviewer: If the extra spaces aren't part of the AWS UI, they're prob not needed. I also recommended a minor ordering change in the second sentence because the original could be interpreted as "the instructions give the impression of credentials being automatically created." This also applies across the other updated areas if you choose to accept it. 

```suggestion
.. Option 1: Direct access keys/CloudFormation (Recommended). Under **Preferred method**, select **Direct access keys**. Expand the **Steps to Generate AWS Account Credentials** section, then follow the displayed instructions to automatically create the necessary credentials using CloudFormation.
```

You can set up CSPM for AWS either by enrolling a single cloud account, or by enrolling an organization containing multiple accounts. Either way, first you will add the CSPM integration, then enable cloud account access.

[discrete]
[[cspm-aws-agentless]]

> == Agentless deployment option

Reviewer 2: I would suggest adding a header for agentless deployment option and capture the methods supported under that section. Also, as per the beta guide shared, you could also add some description of the agentless method and when user would choose this.

Author: I don't want to explain the entire authentication methods twice on the same page (once for agentless and once for agent-based), but I added more detail about the steps other than authentication to this section, and link to the authentication methods.

beta::[]

> The steps to create an agentless deployment are similar to those to create an agent-based deployment. To deploy using agentless technology, follow the instructions below with the following modifications:

Reviewer 2: This is not correct statement, as agentless deployment is simplifying the integration setup. And Cloudformation option in case of agentless is about creation of a role only and in case of agent-based it is creation of the ec2 instance with agent installed on it. For simplicity I would propose to keep the header for agentless and agent based deployment options and detail out the steps in each of these sections.

. Click *Add Cloud Security Posture Management (CSPM)*.
. Select *AWS*, then either *AWS Organization* to onboard multiple accounts, or *Single Account* to onboard an individual account.
. Give your integration a name that matches the purpose or team of the AWS account/organization you want to monitor, for example, `dev-aws-account`.

> . (Optional) click **Advanced settings** to deploy the integration using agentless technology (beta). 

Reviewer 3: Fixes typo, and uses a formal [beta admonition](https://github.com/elastic/docs/blob/master/README.asciidoc#using-the-beta-admonition):
```suggestion
. beta:[] (Optional) Click **Advanced settings** to deploy the integration using agentless technology. 
```

[discrete]
[[cspm-set-up-cloud-access-section]]
== Set up cloud account access
The CSPM integration requires access to AWS's built-in https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_job-functions.html#jf_security-auditor[`SecurityAudit` IAM policy] in order to discover and evaluate resources in your cloud account. There are several ways to provide access.

For most use cases, the simplest option is to use AWS CloudFormation to automatically provision the necessary resources and permissions in your AWS account. This method, as well as several manual options, are described below.

> NOTE: Agentless deployments (beta) support two authentication methods: <<cspm-use-temp-credentials, temporary keys>> and <<cspm-use-keys-directly, direct access keys>>.

Reviewer 3: I'm a little wary of adding the beta tag too many times in one page, but I think twice is still OK, and this makes the page consistent too:
```suggestion
NOTE: beta:[] Agentless deployments support two authentication methods: <<cspm-use-temp-credentials, temporary keys>> and <<cspm-use-keys-directly, direct access keys>>.
```

.. Click *Update IAM role*.
.. Return to {kib} and <<cspm-finish-manual, finish manual setup>>.

> IMPORTANT: Make sure to deploy the CSPM integration to this EC2 instance. When completing setup in {kib}, in the *Setup Access* section, select *Assume role*. Leave **Role ARN** empty unless you want to specify a role the ((agent)) should assume instead of the default role for your EC2 instance. Click *Save and continue*.

Reviewer 2: In case of agentless this is not applicable.

[discrete]

> [[cspm-use-keys-directly]]

Reviewer 2: Great inputs @Omolola-Akinleye 
@benironside is it possible to cover these in the FAQ for CSPM or you recommend any other place for troubleshooting agentless integrations?

> [[cspm-use-keys-directly]]

Author: I think the FAQ seems like a good option. Let's cover this in our next sync
